File: backend/internal/domain/site.go
package domain

type HomeStat struct {
	Value string `json:"value"`
	Label string `json:"label"`
}

type HomeBenefit struct {
	Title string `json:"title"`
	Text  string `json:"text"`
}

type HomeFAQ struct {
	Question string `json:"question"`
	Answer   string `json:"answer"`
}

type HomeCategoryTile struct {
	Label    string `json:"label"`
	URL      string `json:"url"`
	ImageURL string `json:"image_url"`
}

type HomeCta struct {
	Title          string `json:"title"`
	Text           string `json:"text"`
	Schedule       string `json:"schedule"`
	PrimaryLabel   string `json:"primary_label"`
	PrimaryURL     string `json:"primary_url"`
	SecondaryLabel string `json:"secondary_label"`
	SecondaryURL   string `json:"secondary_url"`
}

type HomeContent struct {
	HeroTitle              string             `json:"hero_title"`
	HeroSubtitle           string             `json:"hero_subtitle"`
	CategoryTiles          []HomeCategoryTile `json:"category_tiles"`
	AboutImageURL          string             `json:"about_image_url"`
	Cta                    HomeCta            `json:"cta"`
	StatsTitle             string             `json:"stats_title"`
	Stats                  []HomeStat         `json:"stats"`
	Benefits               []HomeBenefit      `json:"benefits"`
	FAQ                    []HomeFAQ          `json:"faq"`
	FeaturedGuideSlugs     []string           `json:"featured_guide_slugs"`
	FeaturedExcursionSlugs []string           `json:"featured_excursion_slugs"`
	PopularCitySlugs       []string           `json:"popular_city_slugs"`
}

type FooterLink struct {
	Label string `json:"label"`
	URL   string `json:"url"`
}

type FooterColumn struct {
	Title string       `json:"title"`
	Links []FooterLink `json:"links"`
}

type FooterContent struct {
	Phone       string         `json:"phone"`
	Email       string         `json:"email"`
	Telegram    string         `json:"telegram"`
	Description string         `json:"description"`
	Columns     []FooterColumn `json:"columns"`
	Copyright   string         `json:"copyright"`
}

type LegalPage struct {
	Title    string `json:"title"`
	BodyHTML string `json:"body_html"`
}

type LegalContent struct {
	PrivacyPolicy  LegalPage `json:"privacy_policy"`
	SiteRules      LegalPage `json:"site_rules"`
	PlacementRules LegalPage `json:"placement_rules"`
}

type DestinationCity struct {
	Slug string `json:"slug"`
	Name string `json:"name"`
}

type DestinationGroup struct {
	CountrySlug string            `json:"country_slug"`
	CountryName string            `json:"country_name"`
	Cities      []DestinationCity `json:"cities"`
}

type SiteHomePayload struct {
	Content              HomeContent        `json:"content"`
	FeaturedGuides       []PublicGuideDTO   `json:"featured_guides"`
	FeaturedExcursions   []ExcursionView    `json:"featured_excursions"`
	PopularDestinations  []DestinationGroup `json:"popular_destinations"`
}

type SitePayload struct {
	Home   SiteHomePayload `json:"home"`
	Footer FooterContent   `json:"footer"`
	Legal  LegalContent    `json:"legal"`
}

File: packages/api-client/src/api/types/site.ts
import type { Excursion, PublicGuide } from './catalog'

export type HomeStat = { value: string; label: string }
export type HomeBenefit = { title: string; text: string }
export type HomeFAQ = { question: string; answer: string }
export type HomeCategoryTile = { label: string; url: string; image_url: string }
export type HomeCta = {
  title: string
  text: string
  schedule: string
  primary_label: string
  primary_url: string
  secondary_label: string
  secondary_url: string
}

export type HomeContent = {
  hero_title: string
  hero_subtitle: string
  category_tiles: HomeCategoryTile[]
  about_title: string
  about_text: string
  about_image_url: string
  about_button_label: string
  about_button_url: string
  cta: HomeCta
  stats_title: string
  stats: HomeStat[]
  benefits: HomeBenefit[]
  faq: HomeFAQ[]
  featured_guide_slugs: string[]
  featured_excursion_slugs: string[]
  popular_city_slugs: string[]
}

export type FooterLink = { label: string; url: string }
export type FooterColumn = { title: string; links: FooterLink[] }

export type FooterContent = {
  phone: string
  email: string
  telegram: string
  description: string
  columns: FooterColumn[]
  copyright: string
}

export type LegalPage = { title: string; body_html: string }

export type LegalContent = {
  privacy_policy: LegalPage
  site_rules: LegalPage
  placement_rules: LegalPage
}

export type AboutAudienceItem = {
  title: string
  description: string
}

export type AboutPageContent = {
  hero_eyebrow: string
  hero_title: string
  hero_lead: string
  story: string[]
  belief: string
  audience_title: string
  audience_lead: string
  audience: AboutAudienceItem[]
  disclaimer: string
  mission: string
  tagline: string
  closing: string
}

export type DestinationCity = { slug: string; name: string }
export type DestinationGroup = {
  country_slug: string
  country_name: string
  cities: DestinationCity[]
}

export type SitePayload = {
  home: {
    content: HomeContent
    featured_guides: PublicGuide[]
    featured_excursions: Excursion[]
    popular_destinations: DestinationGroup[]
  }
  footer: FooterContent
  legal: LegalContent
  about: AboutPageContent
  telegram_bot_url?: string
}

export type SiteContentPayload = {
  home: HomeContent
  footer: FooterContent
  legal: LegalContent
  about: AboutPageContent
}
